File: fixtures/scale/dart/large/lib/ciphers/keywordsub/keywordsub_key.dart
class KeywordSubKey {
  final String keyword;

  KeywordSubKey(this.keyword);

  /// Keyword first (duplicates dropped), then the remaining letters in order.
  String mixedAlphabet() {
    final sb = StringBuffer();
    for (final c in '${keyword.toUpperCase()}ABCDEFGHIJKLMNOPQRSTUVWXYZ'.split('')) {
      if (!sb.toString().contains(c)) sb.write(c);
    }
    return sb.toString();
  }

  factory KeywordSubKey.defaultKey() => KeywordSubKey('OBSIDIAN');
}

File: fixtures/scale/kotlin/large/src/cryptobench/ciphers/keywordsub/KeywordSubKey.kt
package cryptobench.ciphers.keywordsub

data class KeywordSubKey(val keyword: String) {

    /** Keyword first (duplicates dropped), then the remaining letters in order. */
    fun mixedAlphabet(): String {
        val sb = StringBuilder()
        for (c in keyword.uppercase() + "ABCDEFGHIJKLMNOPQRSTUVWXYZ") {
            if (c in 'A'..'Z' && !sb.contains(c)) sb.append(c)
        }
        return sb.toString()
    }

    companion object {
        fun default(): KeywordSubKey = KeywordSubKey("OBSIDIAN")
    }
}
